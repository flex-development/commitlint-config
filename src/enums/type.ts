/**
 * @file Enums - Type
 * @module commitlint-config/enums/Type
 */

/**
 * Commit types.
 *
 * @see https://commitlint.js.org/#/reference-rules?id=type-enum
 *
 * @enum {Lowercase<string>}
 */
enum Type {
  BUILD = 'build',
  CHORE = 'chore',
  CI = 'ci',
  DOCS = 'docs',
  FEAT = 'feat',
  FIX = 'fix',
  PERF = 'perf',
  REFACTOR = 'refactor',
  REVERT = 'revert',
  STYLE = 'style',
  TEST = 'test',
  WIP = 'wip'
}

export default Type
