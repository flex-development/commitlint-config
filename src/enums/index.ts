/**
 * @file Entry Point - Enums
 * @module commitlint-config/enums
 */

export { default as PromptKind } from './kind-prompt'
export { default as ReferenceAction } from './reference-action'
export { default as Scope } from './scope'
export { default as Type } from './type'
