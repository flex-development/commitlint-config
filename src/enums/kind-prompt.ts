/**
 * @file Enums - PromptKind
 * @module commitlint-config/enums/PromptKind
 */

import type { PromptName } from '@commitlint/types'

/**
 * Prompt names.
 *
 * @see https://commitlint.js.org/#/reference-prompt
 *
 * @enum {PromptName}
 */
enum PromptKind {
  BODY = 'body',
  BREAKING = 'breaking',
  BREAKING_BODY = 'breakingBody',
  ISSUES = 'issues',
  ISSUES_BODY = 'issuesBody',
  IS_BREAKING = 'isBreaking',
  IS_ISSUE_AFFECTED = 'isIssueAffected',
  SCOPE = 'scope',
  SUBJECT = 'subject',
  TYPE = 'type'
}

export default PromptKind
