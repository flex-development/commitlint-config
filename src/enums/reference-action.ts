/**
 * @file Enums - ReferenceAction
 * @module commitlint-config/enums/ReferenceAction
 */

/**
 * Issue reference action types.
 *
 * @enum {Lowercase<string>}
 */
enum ReferenceAction {
  CLOSE = 'closes',
  FIX = 'fixes',
  RELEASE = 'releases',
  RESOLVE = 'resolves'
}

export default ReferenceAction
