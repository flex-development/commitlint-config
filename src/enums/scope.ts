/**
 * @file Enums - Scope
 * @module commitlint-config/enums/Scope
 */

/**
 * Commit scopes.
 *
 * @see https://commitlint.js.org/#/reference-rules?id=scope-enum
 *
 * @enum {Lowercase<string>}
 */
enum Scope {
  BUILD = 'build',
  CJS = 'cjs',
  CLI = 'cli',
  COMPONENTS = 'components',
  CONFIG = 'config',
  CONTROLLERS = 'controllers',
  DECORATORS = 'decorators',
  DEPS = 'deps',
  DEPS_BUNDLE = 'deps-bundle',
  DEPS_DEV = 'deps-dev',
  DEPS_OPT = 'deps-opt',
  DEPS_PEER = 'deps-peer',
  DTOS = 'dtos',
  ENTITIES = 'entities',
  ENUMS = 'enums',
  ESM = 'esm',
  EXCEPTIONS = 'exceptions',
  EXPORTS = 'exports',
  FACTORIES = 'factories',
  FILTERS = 'filters',
  GITHUB = 'github',
  GUARDS = 'guards',
  HOOKS = 'hooks',
  HYBRID = 'hybrid',
  INSTALL = 'install',
  INTERCEPTORS = 'interceptors',
  INTERFACES = 'interfaces',
  INTERNAL = 'internal',
  LIB = 'lib',
  LOADER = 'loader',
  MIDDLEWARE = 'middleware',
  MIXINS = 'mixins',
  MODELS = 'models',
  MODULES = 'modules',
  NAMESPACES = 'namespaces',
  NODE = 'node',
  NVM = 'nvm',
  PATCHES = 'patches',
  PIPES = 'pipes',
  PKG = 'pkg',
  PLUGINS = 'plugins',
  PONYFILL = 'ponyfill',
  PROPS = 'props',
  PROVIDERS = 'providers',
  PUBLIC = 'public',
  REGEX = 'regex',
  RELEASE = 'release',
  RULES = 'rules',
  SCRIPTS = 'scripts',
  SERVICES = 'services',
  SPELLING = 'spelling',
  STORYBOOK = 'storybook',
  STYLES = 'styles',
  SUBDOMAINS = 'subdomains',
  TESTS = 'tests',
  TS = 'ts',
  TS_NEXT = 'ts-next',
  TYPES = 'types',
  UTILS = 'utils',
  VERCEL = 'vercel',
  VITE = 'vite',
  VSCODE = 'vscode',
  WORKFLOWS = 'workflows',
  YARN = 'yarn'
}

export default Scope
