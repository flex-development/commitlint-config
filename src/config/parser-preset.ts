/**
 * @file Config - parserPreset
 * @module commitlint-config/config/parserPreset
 */

import { ReferenceAction } from '#src/enums'
import type { ParserPreset } from '#src/types'
import consola from 'consola'

/**
 * Commit parser preset.
 *
 * Pattern regex:
 *
 * - `headerPattern`: https://regex101.com/r/WAJeLp
 * - `revertPattern`: https://regex101.com/r/78TbQq
 *
 * @const {ParserPreset} parserPreset
 */
const parserPreset: ParserPreset = {
  name: 'conventional-changelog-conventionalcommits',
  parserOpts: {
    commentChar: '#',
    fieldPattern: /^-(?<field>.*?)-(?=\n*)$/m,
    headerCorrespondence: ['type', 'scope', 'breaking', 'subject', 'pr'],
    headerPattern:
      /^(?<type>[a-z]+)(?:\((?<scope>[a-z-]+)\))?(?<breaking>!)?: +(?<subject>(?:.+ \(#(?<pr>\d+)\))|.+)/i,
    issuePrefixes: ['#', 'GH-'],
    issuePrefixesCaseSensitive: true,
    mergeCorrespondence: null,
    mergePattern: null,
    noteKeywords: ['BREAKING-CHANGE'],
    referenceActions: Object.values(ReferenceAction),
    revertCorrespondence: [
      'type',
      'scope',
      'breaking',
      'owner',
      'repository',
      'user',
      'hash'
    ],
    revertPattern:
      /^(?<type>revert)(?:\((?<scope>[a-z-]+)\))?(?<breaking>!)?: (?:|(?:.+\n\n(?:- )?))(?:(?:(?:(?<owner>[\d*a-z~-][\w*.~-]*)\/(?<repository>[\da-z~-][\w.~-]*))|(?<user>[\da-z~-][\w.~-]*))@)?(?<hash>[\da-z]+)/,
    warn: consola.warn.bind(consola)
  }
}

export default parserPreset
