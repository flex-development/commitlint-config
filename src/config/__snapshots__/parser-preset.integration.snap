// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`integration:config/parserPreset > conventional-commits-parser > should parse commits 1`] = `
[
  {
    "body": "Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>",
    "breaking": "!",
    "committerDate": "2023-02-19 01:47:12 -0500",
    "footer": null,
    "gitTags": "",
    "hash": "4b2abcd5d75e9075d0cece6d9b971ca0e64beeb4",
    "header": "revert(deps-dev)!: flex-development/ext-regex@43ced89ca2404e855ab3aedddce445f4e9309ebf",
    "mentions": [
      "43ced89ca2404e855ab3aedddce445f4e9309ebf",
      "github",
      "users",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [],
    "revert": {
      "breaking": "!",
      "hash": "43ced89ca2404e855ab3aedddce445f4e9309ebf",
      "owner": "flex-development",
      "repository": "ext-regex",
      "scope": "deps-dev",
      "subject": "flex-development/ext-regex@43ced89ca2404e855ab3aedddce445f4e9309ebf",
      "type": "revert",
      "user": null,
    },
    "scope": "deps-dev",
    "shortHash": "4b2abcd",
    "subject": "flex-development/ext-regex@43ced89ca2404e855ab3aedddce445f4e9309ebf",
    "type": "revert",
  },
  {
    "body": "Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>",
    "breaking": null,
    "committerDate": "2023-02-18 21:58:00 +0000",
    "footer": null,
    "gitTags": "",
    "hash": "ecafaf2f1431cecf1d326fe7cad3cb0bbaf30cfb",
    "header": "revert(deps-dev): flexdevelopment@ff495037a1a5537bf386530ca540c4c073b47c79",
    "mentions": [
      "ff495037a1a5537bf386530ca540c4c073b47c79",
      "github",
      "users",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [],
    "revert": {
      "breaking": null,
      "hash": "ff495037a1a5537bf386530ca540c4c073b47c79",
      "owner": null,
      "repository": null,
      "scope": "deps-dev",
      "subject": "flexdevelopment@ff495037a1a5537bf386530ca540c4c073b47c79",
      "type": "revert",
      "user": "flexdevelopment",
    },
    "scope": "deps-dev",
    "shortHash": "ecafaf2",
    "subject": "flexdevelopment@ff495037a1a5537bf386530ca540c4c073b47c79",
    "type": "revert",
  },
  {
    "body": "Signed-off-by: Lexus Drumgold <unicornware@flexdevelopment.llc>",
    "breaking": null,
    "committerDate": "2023-02-18 17:00:24 -0500",
    "footer": null,
    "gitTags": "",
    "hash": "cb2eca7ec86eaa45bc89e61d7a3fbfbb7a695c0f",
    "header": "revert: be8b47e536c91bb0940b7bcfb0aae9cd53f794bd",
    "mentions": [
      "flexdevelopment",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [],
    "revert": {
      "breaking": null,
      "hash": "be8b47e536c91bb0940b7bcfb0aae9cd53f794bd",
      "owner": null,
      "repository": null,
      "scope": null,
      "subject": "be8b47e536c91bb0940b7bcfb0aae9cd53f794bd",
      "type": "revert",
      "user": null,
    },
    "scope": null,
    "shortHash": "cb2eca7",
    "subject": "be8b47e536c91bb0940b7bcfb0aae9cd53f794bd",
    "type": "revert",
  },
  {
    "body": "Signed-off-by: Lexus Drumgold <unicornware@flexdevelopment.llc>",
    "breaking": null,
    "committerDate": "2023-02-18 16:45:34 -0500",
    "footer": null,
    "gitTags": "",
    "hash": "0c1a3f9ca18806d9ec06504fd5fff8fc55c4568e",
    "header": "build(deps-dev): bump @flex-development/mkbuild from 1.0.0-alpha.11 to 1.0.0-alpha.13",
    "mentions": [
      "flex-development",
      "flexdevelopment",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [],
    "revert": null,
    "scope": "deps-dev",
    "shortHash": "0c1a3f9",
    "subject": "bump @flex-development/mkbuild from 1.0.0-alpha.11 to 1.0.0-alpha.13",
    "type": "build",
  },
  {
    "body": null,
    "breaking": null,
    "committerDate": "2023-02-18 16:37:45 -0500",
    "footer": "- codecov/codecov-bash#83
- https://github.com/codecov/codecov-bash/issues/83#issuecomment-997761125
- https://docs.github.com/actions/using-workflows/events-that-trigger-workflows#pull_request

Signed-off-by: Lexus Drumgold <unicornware@flexdevelopment.llc>",
    "gitTags": "",
    "hash": "0d60f394051cda7368c5dced0fccd794e16e395c",
    "header": "ci(workflows): [\`ci\`] use \`github.event.pull_request.head.sha\` for codecov",
    "mentions": [
      "flexdevelopment",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [
      {
        "action": null,
        "issue": "83",
        "owner": "codecov",
        "prefix": "#",
        "raw": "- codecov/codecov-bash#83",
        "repository": "codecov-bash",
      },
      {
        "action": null,
        "issue": "issuecomment-997761125",
        "owner": "",
        "prefix": "#",
        "raw": "- https://github.com/codecov/codecov-bash/issues/83#issuecomment-997761125",
        "repository": "/github.com/codecov/codecov-bash/issues/83",
      },
    ],
    "revert": null,
    "scope": "workflows",
    "shortHash": "0d60f39",
    "subject": "[\`ci\`] use \`github.event.pull_request.head.sha\` for codecov",
    "type": "ci",
  },
  {
    "body": "Signed-off-by: Lexus Drumgold <unicornware@flexdevelopment.llc>",
    "breaking": null,
    "committerDate": "2023-02-15 18:32:31 -0500",
    "footer": null,
    "gitTags": " (tag: 1.0.0)",
    "hash": "4a04c8afb2b4ba1dfea3ff11e7a14d2850896e2d",
    "header": "release: 1.0.0 (#3)",
    "mentions": [
      "flexdevelopment",
    ],
    "merge": null,
    "notes": [],
    "pr": "3",
    "references": [
      {
        "action": null,
        "issue": "3",
        "owner": null,
        "prefix": "#",
        "raw": "release: 1.0.0 (#3",
        "repository": null,
      },
    ],
    "revert": null,
    "scope": null,
    "shortHash": "4a04c8a",
    "subject": "1.0.0 (#3)",
    "type": "release",
  },
  {
    "body": "Signed-off-by: Lexus Drumgold <unicornware@flexdevelopment.llc>",
    "breaking": null,
    "committerDate": "2023-02-15 04:05:56 -0500",
    "footer": null,
    "gitTags": "",
    "hash": "126e1b2d7d8ab5d2629718efc567cf96f29a8819",
    "header": "docs: api",
    "mentions": [
      "flexdevelopment",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [],
    "revert": null,
    "scope": null,
    "shortHash": "126e1b2",
    "subject": "api",
    "type": "docs",
  },
  {
    "body": "- https://docs.codecov.com/docs/manual-critical-file-labelling

Signed-off-by: Lexus Drumgold <unicornware@flexdevelopment.llc>",
    "breaking": null,
    "committerDate": "2023-02-15 00:21:49 -0500",
    "footer": null,
    "gitTags": "",
    "hash": "8e2432c3cfb9adfb5c725d120a1144096a11e69d",
    "header": "chore(tests): [codecov] label critical files",
    "mentions": [
      "flexdevelopment",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [],
    "revert": null,
    "scope": "tests",
    "shortHash": "8e2432c",
    "subject": "[codecov] label critical files",
    "type": "chore",
  },
  {
    "body": "Signed-off-by: Lexus Drumgold <unicornware@flexdevelopment.llc>",
    "breaking": null,
    "committerDate": "2023-02-15 00:09:30 -0500",
    "footer": null,
    "gitTags": "",
    "hash": "eb850cff64d86eadbcb238b3af7cae702b7ab405",
    "header": "feat(utils): \`extRegex\`",
    "mentions": [
      "flexdevelopment",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [],
    "revert": null,
    "scope": "utils",
    "shortHash": "eb850cf",
    "subject": "\`extRegex\`",
    "type": "feat",
  },
  {
    "body": "Signed-off-by: Lexus Drumgold <unicornware@flexdevelopment.llc>",
    "breaking": null,
    "committerDate": "2023-02-14 15:24:18 -0500",
    "footer": null,
    "gitTags": "",
    "hash": "d8fe2fb256564dda77a1cb2fac78804a709e3e21",
    "header": "ci(workflows): [\`ci\`] upload coverage report to codecov",
    "mentions": [
      "flexdevelopment",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [],
    "revert": null,
    "scope": "workflows",
    "shortHash": "d8fe2fb",
    "subject": "[\`ci\`] upload coverage report to codecov",
    "type": "ci",
  },
  {
    "body": "Signed-off-by: Lexus Drumgold <unicornware@flexdevelopment.llc>",
    "breaking": null,
    "committerDate": "2023-02-14 15:11:12 -0500",
    "footer": null,
    "gitTags": "",
    "hash": "c7dfaab6fec3eba06d486b11bd6d9f1280c5b3f6",
    "header": "ci: [@dependabot] add config",
    "mentions": [
      "dependabot",
      "flexdevelopment",
    ],
    "merge": null,
    "notes": [],
    "pr": null,
    "references": [],
    "revert": null,
    "scope": null,
    "shortHash": "c7dfaab",
    "subject": "[@dependabot] add config",
    "type": "ci",
  },
]
`;
